dev/tasks/tasks.yml

```diff
@@ -1256,12 +1256,12 @@ tasks:
         GO: 1.15
       run: debian-go
 
-  test-ubuntu-20.10-docs:
+  test-ubuntu-20.04-docs:
     ci: azure
     template: docker-tests/azure.linux.yml
     params:
       env:
-        UBUNTU: "20.10"
+        UBUNTU: "20.04"
```

Comment:
`test-ubuntu-default-docs` already uses Ubuntu 20.04.

Reply:
Good point! Given that this additional job was introduced in https://github.com/apache/arrow/pull/10430 and things have moved on since then, I've removed the extra job.  Or should I add it back in and pin it to a different version? It'd be good to know your thoughts too @jonkeane as you were the author of the PR I mentioned above and I don't fully understand everything that's going on there.

Reply:
IIRC, I introduced `test-ubuntu-default-docs` because we started being unable to build docs on 20.04 (and that inability wasn't noticed cause we were only testing 20.10 before this) 

It looks like the 20.10 build was changed from a much older one in https://github.com/apache/arrow/pull/9933.

I'm not sure what the purpose of having a default build and then a separate build on a non-default version is. The one reason I can think of is that we want the non-default version to be a test of a new(ish) ubuntu version, to check for issues there. If that is the case (and we want to maintain that), we probably should build on 21.04 or 21.10. But I'm also totally fine removing the non-default docs build if it's more hassle than is helpful to maintain it.

Reply:
OK, sounds like it made sense at the time but no longer serves a specific purpose, so unless there's any other reason to keep it in, let's leave it out for now (and add it back in if necessary later).

Reply:
We chose Ubuntu 20.10 to reduce build time by using system gRPC: https://github.com/apache/arrow/pull/9933#pullrequestreview-630895705

I'm OK with removing the docs job with non default `UBUNTU` version.

Could you also remove `UBUNTU=20.10` from `dev/release/post-09-docs.sh` and `ci/docker/ubuntu-20.10-cpp.dockerfile` file?

Reply:
Aha, thanks for clarifying that @kou!  Just to check my understanding, did you mean remove the reference to 20.10 in `dev/release/post-09-docs.sh` but entirely remove the file `ci/docker/ubuntu-20.10-cpp.dockerfile` as no other tasks depend on this now?

Reply:
Yes.